iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -77,6 +77,14 @@ cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitia
         return cxx::error<ThreadError>(Thread::errnoToEnum(createResult.get_error().errnum));
     }
 
+    posixCall(iox_pthread_setname_np)(uninitializedThread->m_threadHandle, m_name.c_str())
+        .successReturnValue(0)
+        .evaluate()
+        .expect("This should never happen! Failed to set thread name.");
+    /// @todo thread specific comm file under /proc/self/task/[tid]/comm is read. Opening this file can fail
```

Comment:
Nitpick: Maybe link an issue if this todo stays.

Reply:
According to https://man7.org/linux/man-pages/man3/pthread_setname_np.3.html, a failing `open` call may influence the return value of `pthread_setname_np`.

Reply:
@FerdinandSpitzschnueffler I suppose we should add a note in the docs somewhere about `procfs`. Did you check the QNX manual?

Reply:
@mossmaurice I think we should be platform independent here. Additionally, I tend to refactor the `getName` and `setName` methods so that we don't terminate in an error case but just log that setting the name didn't work - maybe we don't even have to distinguish the failure causes. But I would like to postpone it to another PR when the free functions will be also removed.